frontend/packages/knative-plugin/src/components/overview/__tests__/RevisionsOverviewList.spec.tsx

```diff
@@ -0,0 +1,89 @@
+import * as utils from '@console/internal/components/utils';
+import RevisionsOverviewList from '../RevisionsOverviewList';
+import RevisionsOverviewListItem from '../RevisionsOverviewListItem';
+import { mockRevisionsData, mockServiceData } from '../__mocks__/overview-knative-mock';
+
+describe('RevisionsOverviewList', () => {
+  it('should component exists', () => {
+    const wrapper = shallow(<RevisionsOverviewList revisions={[]} service={mockServiceData} />);
+    expect(wrapper.exists()).toBeTruthy();
+  });
+
+  it('should have title Revisions', () => {
+    const wrapper = shallow(
+      <RevisionsOverviewList revisions={mockRevisionsData} service={mockServiceData} />,
+    );
+    expect(wrapper.find(utils.SidebarSectionHeading)).toHaveLength(1);
+    expect(
+      wrapper
+        .find(utils.SidebarSectionHeading)
+        .at(0)
+        .props().text,
+    ).toEqual('Revisions');
+  });
+
+  it('should show info if no Revisions present and traffic split sshould button should be disabled', () => {
+    const spyUseAccessReview = jest.spyOn(utils, 'useAccessReview');
+    spyUseAccessReview.mockReturnValue(true);
+    const wrapper = shallow(<RevisionsOverviewList revisions={[]} service={mockServiceData} />);
+    expect(wrapper.find('span')).toHaveLength(1);
+    expect(
+      wrapper
+        .find('span')
+        .at(0)
+        .props().className,
+    ).toEqual('text-muted');
+    expect(
+      wrapper
+        .find('span')
+        .at(0)
+        .props().children,
+    ).toEqual('No Revisions found for this resource.');
+    expect(
+      wrapper
+        .find(Button)
+        .at(0)
+        .props().isDisabled,
+    ).toBe(true);
+  });
+
+  it('should have button for traffic distribution and enabled', () => {
+    const spyUseAccessReview = jest.spyOn(utils, 'useAccessReview');
+    spyUseAccessReview.mockReturnValue(true);
+    const wrapper = shallow(
+      <RevisionsOverviewList revisions={mockRevisionsData} service={mockServiceData} />,
+    );
+    expect(wrapper.find(Button)).toHaveLength(1);
+    expect(
+      wrapper
+        .find(Button)
+        .at(0)
+        .props().children,
+    ).toEqual('Set Traffic Distribution');
+    expect(
+      wrapper
+        .find(Button)
+        .at(0)
+        .props().isDisabled,
+    ).toBe(false);
+  });
+
+  it('should not show button for traffic distribution if access is not there', () => {
+    const spyUseAccessReview = jest.spyOn(utils, 'useAccessReview');
+    spyUseAccessReview.mockReturnValue(false);
+    const wrapper = shallow(
+      <RevisionsOverviewList revisions={mockRevisionsData} service={mockServiceData} />,
+    );
+    expect(wrapper.find(Button).exists()).toBe(false);
+  });
+
+  it('should render RevisionsOverviewListItem', () => {
+    const wrapper = shallow(
+      <RevisionsOverviewList revisions={mockRevisionsData} service={mockServiceData} />,
+    );
+    expect(wrapper.find(RevisionsOverviewListItem)).toHaveLength(1);
+  });
+});
```

Comment:
Can you also add a test case for ```onClick``` on ```Set Traffic Distribution``` button?

Reply:
Added